### PR TITLE
EVG-2002: improve cancelation for runner operations

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: hostinit/setup.go [hunk 1]
```diff
@@ -396,7 +412,8 @@ func (init *HostInit) copyScript(target *host.Host, name, script string) error {
 	grip.Infof("Directories command: '%#v'", makeDirectoryCmd)
 
 	// run the make shell command with a timeout
-	ctx, cancel := context.WithTimeout(context.TODO(), SCPTimeout)
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, SCPTimeout)
```
Comment (bsamek) [inline hostinit/setup.go]: You can't `ctx, cancel := context.WithTimeout(ctx, SCPTimeout)`?

File: hostinit/setup.go [hunk 2]
```diff
@@ -420,7 +437,7 @@ func (init *HostInit) copyScript(target *host.Host, name, script string) error {
 		Options:        append([]string{"-P", hostInfo.Port}, sshOptions...),
 	}
 	// run the command to scp the agent with a timeout
-	ctx, cancel = context.WithTimeout(context.TODO(), SCPTimeout)
+	ctx, cancel = context.WithTimeout(ctx, SCPTimeout)
```
Comment (bsamek) [inline hostinit/setup.go]: This duplicates 416.

File: hostinit/setup.go [hunk 3]
```diff
@@ -588,7 +605,8 @@ func (init *HostInit) LoadClient(target *host.Host) (*LoadClientResult, error) {
 	scpOut := &util.CappedWriter{&bytes.Buffer{}, 1024 * 1024}
 
 	// run the make shell command with a timeout
-	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, 30*time.Second)
```
Comment (bsamek) [inline hostinit/setup.go]: This reads to me like we're creating `CancelFunc`s and deferring them on blocking calls (here and later in this function). What is the purpose of this?

File: hostinit/setup.go [hunk 4]
```diff
@@ -653,7 +671,7 @@ func (init *HostInit) LoadClient(target *host.Host) (*LoadClientResult, error) {
 	}, nil
 }
 
-func (init *HostInit) fetchRemoteTaskData(taskId, cliPath, confPath string, target *host.Host) error {
+func (init *HostInit) fetchRemoteTaskData(ctx context.Context, taskId, cliPath, confPath string, target *host.Host) error {
```
Comment (bsamek) [inline hostinit/setup.go]: This function does use its passed context. Should it (line 709)?

File: hostinit/setup.go (hunk 1 above)
Comment (tychoish) [inline hostinit/setup.go]: to avoid shadowing errors.

File: hostinit/setup.go (hunk 2 above)
Comment (tychoish) [inline hostinit/setup.go]: we want to create a new timeout for the second operation

File: hostinit/setup.go (hunk 3 above)
Comment (tychoish) [inline hostinit/setup.go]: the defer is just for cleanup in case/when we finish before the runout.

File: hostinit/setup.go (hunk 4 above)
Comment (tychoish) [inline hostinit/setup.go]: fixed

**Pull request merged**